src/inc/til/env.h

```diff
@@ -0,0 +1,575 @@
+            return result;
+        }
+
+        struct wstring_case_insensitive_compare
+        {
+            [[nodiscard]] bool operator()(const std::wstring& lhs, const std::wstring& rhs) const noexcept
+            {
+                return compare_string_ordinal(lhs, rhs) == CSTR_LESS_THAN;
+            }
+        };
+
+        namespace vars
+        {
+            static constexpr std::wstring_view system_root{ L"SystemRoot" };
+            static constexpr std::wstring_view system_drive{ L"SystemDrive" };
+            static constexpr std::wstring_view all_users_profile{ L"ALLUSERSPROFILE" };
+            static constexpr std::wstring_view public_var{ L"PUBLIC" };
+            static constexpr std::wstring_view program_data{ L"ProgramData" };
+            static constexpr std::wstring_view computer_name{ L"COMPUTERNAME" };
+            static constexpr std::wstring_view user_name{ L"USERNAME" };
+            static constexpr std::wstring_view user_domain{ L"USERDOMAIN" };
+            static constexpr std::wstring_view user_dns_domain{ L"USERDNSDOMAIN" };
+            static constexpr std::wstring_view home_drive{ L"HOMEDRIVE" };
+            static constexpr std::wstring_view home_share{ L"HOMESHARE" };
+            static constexpr std::wstring_view home_path{ L"HOMEPATH" };
+            static constexpr std::wstring_view user_profile{ L"USERPROFILE" };
+            static constexpr std::wstring_view app_data{ L"APPDATA" };
+            static constexpr std::wstring_view local_app_data{ L"LOCALAPPDATA" };
+
+            static constexpr std::wstring_view program_files{ L"ProgramFiles" };
+            static constexpr std::wstring_view program_files_x86{ L"ProgramFiles(x86)" };
+            static constexpr std::wstring_view program_files_arm64{ L"ProgramFiles(Arm)" };
+            static constexpr std::wstring_view program_w6432{ L"ProgramW6432" };
+            static constexpr std::wstring_view common_program_files{ L"CommonProgramFiles" };
+            static constexpr std::wstring_view common_program_files_x86{ L"CommonProgramFiles(x86)" };
+            static constexpr std::wstring_view common_program_files_arm64{ L"CommonProgramFiles(Arm)" };
+            static constexpr std::wstring_view common_program_w6432{ L"CommonProgramW6432" };
+
+            const std::map<std::wstring, std::wstring_view> program_files_map{
+                { L"ProgramFilesDir", program_files },
+                { L"CommonFilesDir", common_program_files },
+#ifdef _WIN64
+#ifdef _M_ARM64
+                { L"ProgramFilesDir (Arm)", program_files_arm64 },
+                { L"CommonFilesDir (Arm)", common_program_files_arm64 },
+#endif
+                { L"ProgramFilesDir (x86)", program_files_x86 },
+                { L"CommonFilesDir (x86)", common_program_files_x86 },
+                { L"ProgramW6432Dir", program_w6432 },
+                { L"CommonW6432Dir", common_program_w6432 },
+#endif
+            };

```

Comment:
Since this is lacking both `inline` and `static` this would create a linker error if included in more than one `.cpp`. A better idea would be to move this into the function that uses it. `static` variables inside functions are lazily created (= only when the function is first called).

But looking at `get_program_files()` it doesn't seem like the map is being used as a map anyways? Nor is the `std::wstring` key being used as a string? So you could just write this instead:
```suggestion
            const std::array program_files_map{
                std::pair{ L"ProgramFilesDir", program_files },
                std::pair{ L"CommonFilesDir", common_program_files },
#ifdef _WIN64
#ifdef _M_ARM64
                std::pair{ L"ProgramFilesDir (Arm)", program_files_arm64 },
                std::pair{ L"CommonFilesDir (Arm)", common_program_files_arm64 },
#endif
                std::pair{ L"ProgramFilesDir (x86)", program_files_x86 },
                std::pair{ L"CommonFilesDir (x86)", common_program_files_x86 },
                std::pair{ L"ProgramW6432Dir", program_w6432 },
                std::pair{ L"CommonW6432Dir", common_program_w6432 },
#endif
            };
```

Then use it as:
```cpp
                for (const auto& [keyName, varName] : til::details::vars::program_files_map)
                {
                    auto value = til::details::wil_env::RegQueryValueExW<std::wstring, 256>(key.get(), keyName);
                    set_user_environment_var(varName, value);
                }
```